web_monitoring/html_diff_render.py

```diff
@@ -810,17 +926,51 @@ def html(self):
 # I had some weird concern that I needed to make this token a single word with
 # no spaces, but now that I know this differ more deeply, this is pointless.
 class ImgTagToken(tag_token):
-    def __new__(cls, tag, data, html_repr, pre_tags=None,
+
+    regex = re.compile(r"(src|srcset)=('|\")(.*?)('|\")")
+
+    def are_any_urls_equal(self, other):
+        strict_urls = self.strict_urls
+        self_array = self[7:-3]
+        self_array = self_array.split(',')
+        other_array = other[7:-3]
+        other_array = other_array.split(',')
+        for url_a in self_array:
+            for url_b in other_array:
+                if strict_urls:
+                    return StrictUrlRule.clean_resource_url(url_a[1:-1], url_b[1:-1], strict_urls)
+                elif url_a == url_b:
+                    return True
+        return False
+
+    def __new__(cls, tag, data, html_repr, strict_urls, pre_tags=None,
                 post_tags=None, trailing_whitespace=""):
-        obj = DiffToken.__new__(cls, "\n\nImg:%s\n\n" % data,
+        found_urls = ImgTagToken.regex.findall(html_repr)
+        urls = []
+        for url in found_urls:
+            if url[0] == 'srcset':
+                if ' ' in url[2]:
+                    urls.append(url[2].split(' ')[0])
+                    continue
+            urls.append(url[2])
+        obj = DiffToken.__new__(cls, "\n\nImg:%s\n\n" % str(urls),
                             pre_tags=pre_tags,
                             post_tags=post_tags,
                             trailing_whitespace=trailing_whitespace)
         obj.tag = tag
         obj.data = data
         obj.html_repr = html_repr
+        obj.strict_urls = strict_urls
         return obj
 
+    def __eq__(self, other):
+        if isinstance(other, ImgTagToken):
+            return self.are_any_urls_equal(other)
+        return False
+
+    def __hash__(self):
+        return super.__hash__(self)
```

Comment:
I think you can just leave this method out; the effect will be the same.

Reply:
If I don't include this I get a `TypeError: unhashable type: 'ImgTagToken'`

Reply:
Ah! I hadn’t realized Python has some special behavior when `__eq__` is defined. From [the docs](https://docs.python.org/3/reference/datamodel.html#object.__hash__):

> A class that overrides `__eq__()` and does not define `__hash__()` will have its `__hash__()` implicitly set to `None`… If a class that overrides `__eq__()` needs to retain the implementation of `__hash__()` from a parent class, the interpreter must be told this explicitly by setting `__hash__ = <ParentClass>.__hash__`.

So I guess you do need to implement `__hash__`, but it might be better to use the suggested for above, or to at least change `super.__hash__(self)` to use the more standard `super().__hash__()` form.